### PR TITLE
Adding `useWillChange` to apply `will-change` to animating values

### PR DESCRIPTION
**Pull request by mattgperry**

### CODE REVIEW

File: dev/examples/Prop-style.tsx [hunk 1]
```diff
@@ -15,7 +15,7 @@ const style = {
 
 export const App = () => {
     return (
-        <motion.div static>
+        <motion.div>
```
Comment (mattgperry) [inline dev/examples/Prop-style.tsx:18]: `static` no longer exists as a prop - change unrelated to PR

File: packages/framer-motion/src/render/utils/animation-state.ts [hunk 2]
```diff
@@ -82,10 +80,6 @@ export function createAnimationState(
         return acc
     }
 
-    function isAnimated(key: string) {
```
Comment (mattgperry) [inline packages/framer-motion/src/render/utils/animation-state.ts:85]: Unused API

File: packages/framer-motion/src/value/use-will-change/__tests__/index.test.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                    expect(container.firstChild).toHaveStyle(
+                        "will-change: transform, background-color;"
+                    )
+                    resolve()
+                })
+            })
+        })
+    })
+
+    test("Removes `transform` when all transforms finish animating", async () => {
```
Comment (shuangq) [inline packages/framer-motion/src/value/use-will-change/__tests__/index.test.tsx:63]: We could maybe cover one more test to verify that when all the motion values (not just transforms) finish animating, will change is revert to "auto"?

**Review by shuangq — APPROVED** [review]
Left some questions, but code looks good in general 👍

File: packages/framer-motion/src/value/use-will-change/index.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        } else if (
+            !isTransformOriginProp(name) &&
+            !isCSSVariable(name) &&
+            name !== "willChange"
+        ) {
+            memberName = camelToDash(name)
+        }
+
+        if (memberName) {
+            addUniqueItem(this.members, memberName)
```
[the remark this answers is not in the document]
Comment (mattgperry) [reply packages/framer-motion/src/value/use-will-change/index.ts:31]: It's more just so I can use `join` more conveniently when we make the `will-change` string

File: packages/framer-motion/src/utils/array.ts [hunk 5]
```diff
@@ -1,10 +1,16 @@
 export function addUniqueItem<T>(arr: T[], item: T) {
-    arr.indexOf(item) === -1 && arr.push(item)
+    if (arr.indexOf(item) === -1) {
+        arr.push(item)
+        return true
```
[the remark this answers is not in the document]
Comment (mattgperry) [reply packages/framer-motion/src/utils/array.ts:4]: This is actually unnecessary now. I was using to to return from the function whether the operation was successful. But not using this functionality any more so I can take it out

**Pull request merged**